praw/reddit.py

```diff
@@ -542,6 +556,60 @@ def generator(url):
 
         return generator(url)
 
+    def _objectify_request(
+        self,
+        data: Optional[
+            Union[Dict[str, Union[str, Any]], bytes, IO, str]
+        ] = None,
+        files: Optional[Dict[str, IO]] = None,
+        method: str = "",
+        params: Optional[Union[str, Dict[str, str]]] = None,
+        path: str = "",
+    ) -> Any:
+        """Run a request through the ``Objector``.
+
+        :param data: Dictionary, bytes, or file-like object to send in the body
+            of the request (default: None).
+        :param files: Dictionary, filename to file (like) object mapping
+            (default: None).
+        :param method: The HTTP method (e.g., GET, POST, PUT, DELETE).
+        :param params: The query parameters to add to the request (default:
+            None).
+        :param path: The path to fetch.
+
+        """
+        data = self.request(
+            data=data, files=files, method=method, params=params, path=path
+        )
+        return self._objector.objectify(data)
+
+    def _handle_rate_limit(
+        self,
+        exception: RedditAPIException,
+        retry: Callable[[Any], Any],
+        *retry_args: Any,
+        **retry_kwargs: Any
+    ) -> Any:
+        for item in exception.items:
+            if "RATELIMIT" == item.error_type:
+                amount_search = search(
+                    r"([0-9]{1,2}) (seconds?|minutes?)", item.message
+                )
+                if not amount_search:
+                    break
+                seconds = int(amount_search.group(1))
+                if "minute" in amount_search.group(2):
+                    seconds *= 60
+                if seconds <= int(self.config.ratelimit_seconds):
+                    sleep_seconds = seconds + min(seconds / 10, 1)
+                    logger.debug(
+                        "Rate limit hit, sleeping for %.2f " "seconds",
```

Comment:
Can you pass in a format string here instead? Also looks like black concatenated your strings.

Reply:
It's a logger string. Logging doesn't use format strings.

Reply:
I understand that logging supports this format, as does print if I'm not mistaken. I personally think it's cleaner and more consistent to pre-format strings before passing them in.